azext_iot/common/base_discovery.py

```diff
@@ -179,10 +179,17 @@ def find_resource(self, resource_name: str, rg: str = None):
         resource_list = self.get_resources()
 
         if resource_list:
-            target = next(
-                (resource for resource in resource_list if resource_name.lower() == resource.name.lower()),
-                None
-            )
+            target = None
+            if isinstance(resource_list[0], dict):
```

Comment:
minor nit: we could use a placeholder value like:
`name = resource.get('name', '') if isinstance(resource_list[0], dict) else resource.name`

and then reference it as `name.lower()` later - just to avoid rewriting the iteration here, but functionally there's no difference.

Reply:
discussed - not proceeding with this since it is more complicated than this